app/routes/application.js

```diff
@@ -0,0 +1,28 @@
+import Route from "@ember/routing/route";
+
+export default class ApplicationRoute extends Route {
+  heightChangeResizeObserver = null;
+
+  constructor() {
+    super(...arguments);
+    this.setupHeightChangeDetection();
+  }
+
+  setupHeightChangeDetection() {
+    const resizeObserver = new ResizeObserver((entries) => {
+      let body = entries[0].target;
+      let newHeight = body.clientHeight;
+      window.parent.postMessage({ height: newHeight });
```

Comment:
The hosting application should be able to do something like this:

```js
      let iframe = document.querySelector("iframe");
      window.addEventListener("message", (event) => {
        // TODO: detect that the message came from the complaint form app
        iframe.height = event.data?.height;
      });
```

I tested this locally and it seems to work, but I haven't tried in cross-origin setups.